internal/packages/packagecontent/topackage.go

```diff
@@ -4,16 +4,86 @@ import (
 	"bytes"
 	"context"
 	"path/filepath"
+	"regexp"
+	"sort"
 	"strings"
 
+	manifestsv1alpha1 "package-operator.run/apis/manifests/v1alpha1"
+
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/yaml"
 
 	"package-operator.run/internal/packages"
 )
 
-func PackageFromFiles(ctx context.Context, scheme *runtime.Scheme, files Files) (pkg *Package, err error) {
+var componentFileRE = regexp.MustCompile("^([a-z][a-z0-9-]{0,62})/(.+)$")
```

Comment:
Where does this regex come from? could you document it? for example, why are not dots allowed in the first path part?

Reply:
You made me realize I used the wrong regex. Now I updated it and added a comment to explain it.

Reply:
@eqrx PTAL if the comment is clear enough